mplc/partner.py

```diff
@@ -58,35 +72,56 @@ def _decorator(self, *args,
     def num_labels(self):
         return self.y_train.shape[1]
 
+    @property
+    def data_volume(self):
+        return len(self.y_train)
+
+    def corrupt(self):
+        if self.corruption_method == 'not_corrupted':
+            logger.debug(f"   Partner #{self.id}: not corrupted.")
+        else:
+            logger.debug(
+                f"... Corrupting ({self.corruption_method}) "
+                f" {self.proportion_corrupted * 100} "
+                f"percent of the data of partner #{self.id}")
+            if self.corruption_method == 'duplication':
+                if not self.duplicated_partner:
+                    raise AttributeError(f'The partner for duplication is missing, please set the '
+                                         f'.duplicated_partner attribute to the partner '
+                                         f'with id {self.duplicated_partner_id}')
```

Comment:
What do you think of picking a random partner when the `duplicated_partner_id` is missing ?

Reply:
We do pick a random partner amongst those with enough data
It's done in the scenario